src/relay/op/tensor/transform.cc

```diff
@@ -454,6 +454,61 @@ RELAY_REGISTER_OP("full")
 .set_support_level(3)
 .add_type_rel("Full", FullRel);
 
+bool TensorRel(const Array<Type>& types,
+               int num_inputs,
+               const Attrs& attrs,
+               const TypeReporter& reporter) {
+  CHECK_EQ(types.size(), 1);
+  const TensorAttrs* param = attrs.as<TensorAttrs>();
+
+  reporter->Assign(types[0], TensorTypeNode::make(param->shape, param->dtype));
+  return true;
+}
+
+Expr MakeZeros(Array<IndexExpr> shape,
+               DataType dtype) {
+  auto attrs = make_node<TensorAttrs>();
+  attrs->shape = std::move(shape);
+  attrs->dtype = std::move(dtype);
```

Comment:
I feel that we don't really need to set `shape` and `dtype` here, right?

Reply:
It is necessary as it is the only info that help the op to infer - it has no input argument!

Reply:
hmm, sorry. I didn't notice that...